sprint0/sprint0/Input/Controller/KeyboardItemController.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Xna.Framework.Input;
+
+namespace sprint0
+{
+    class KeyboardItemController : IController
+    {
+        private readonly Game1 game;
+        private readonly Dictionary<Keys, ICommand> controllerMappings;
+        private Keys[] previousPressedKeys;
+        private readonly HashSet<Keys> movementKeys;
+
+        public KeyboardItemController(Game1 game)
+        {
+            this.game = game;
+            controllerMappings = new Dictionary<Keys, ICommand>();
+            movementKeys = new HashSet<Keys> { Keys.W, Keys.A, Keys.S, Keys.D, Keys.Up, Keys.Down, Keys.Left, Keys.Right };
+            previousPressedKeys = Keyboard.GetState().GetPressedKeys();
+            RegisterCommand(Keys.Q, new QuitCommand(game));
+            RegisterCommand(Keys.R, new ResetCommand(game));
+            RegisterCommand(Keys.C, new CreditsCommand(game));
+            RegisterCommand(Keys.W, new UpCommand(game));
+            RegisterCommand(Keys.S, new DownCommand(game));
+            RegisterCommand(Keys.A, new LeftCommand(game));
+            RegisterCommand(Keys.D, new RightCommand(game));
+            RegisterCommand(Keys.Up, new UpCommand(game));
+            RegisterCommand(Keys.Down, new DownCommand(game));
+            RegisterCommand(Keys.Left, new LeftCommand(game));
+            RegisterCommand(Keys.Right, new RightCommand(game));
+            RegisterCommand(Keys.Z, new SwordCommand(game));
+            RegisterCommand(Keys.N, new SwordCommand(game));
+            RegisterCommand(Keys.D1, new OneCommand(game));
+            RegisterCommand(Keys.D2, new TwoCommand(game));
+            RegisterCommand(Keys.D3, new ThreeCommand(game));
+            RegisterCommand(Keys.D4, new FourCommand(game));
+            RegisterCommand(Keys.E, new PauseCommand(game));
+            RegisterCommand(Keys.Space, new ToggleTestModeCommand(game));
+            RegisterCommand(Keys.B, new BItemCommand(game));
+
```

Comment:
Not sure if i am just missing it but how is this different from KeyboardController and why does it need to be independent?

Reply:
this is part of the keyboard controller. the BItem is specifically for only using the b object. with the normal commands, you can shoot arrow, throw boomerang, etc. even if you're not holding it